src/goLanguageServer.ts

```diff
@@ -143,6 +144,44 @@ export async function registerLanguageFeatures(ctx: vscode.ExtensionContext) {
 						return null;
 					}
 					return next(document, token);
+				},
+				provideCompletionItem: (
+					document: vscode.TextDocument,
+					position: vscode.Position,
+					context: vscode.CompletionContext,
+					token: vscode.CancellationToken,
+					next: ProvideCompletionItemsSignature
+				) => {
+						const editorParamHintsEnabled = vscode.workspace.getConfiguration('editor.parameterHints', document.uri)['enabled'];
```

Comment:
Are these both necessary? I would think that the purpose of the `document.uri` parameter is to decide if the setting applies to the current file.

Reply:
That's what I thought but it didn't work that way. Good news is that [vscode 1.42 api](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/42169/commits/15e3d9ed635760023f0cacdbe0d6710a4c814673) has ConfigurationScope that includes {languageId, uri} pair. If we use it, getConfiguration does the proper job (applying the language specific config on top). Added TODO.